From f936dd32e6d6f87195ee9a5e22237fda1d8aa0cf Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:07:57 -0400
Subject: [tools] perf: Add cat as fallback pager

Message-id: <1410181752-25631-233-git-send-email-jolsa@redhat.com>
Patchwork-id: 91917
O-Subject: [PATCH RHEL7.1 BZ1134356 232/307] perf tools: Add cat as fallback pager
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 21cfc5e10395eba795c0610cf3cf7f0a4f6f33ff
Author: Michael Lentine <mlentine@google.com>
Date: Tue May 20 11:48:49 2014 +0200

description
===========
This patch adds a fallback to cat for the pager. This is useful
on environments, such as Android, where less does not exist.
It is better to default to cat than to abort.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/pager.c b/tools/perf/util/pager.c
index 3322b84..31ee02d 100644
--- a/tools/perf/util/pager.c
+++ b/tools/perf/util/pager.c
@@ -57,13 +57,13 @@ void setup_pager(void)
  }
  if (!pager)
   pager = getenv("PAGER");
- if (!pager) {
-  if (!access("/usr/bin/pager", X_OK))
-   pager = "/usr/bin/pager";
- }
+ if (!(pager || access("/usr/bin/pager", X_OK)))
+  pager = "/usr/bin/pager";
+ if (!(pager || access("/usr/bin/less", X_OK)))
+  pager = "/usr/bin/less";
  if (!pager)
-  pager = "less";
- else if (!*pager || !strcmp(pager, "cat"))
+  pager = "cat";
+ if (!*pager || !strcmp(pager, "cat"))
   return;
 
  spawned_pager = 1; /* means we are emitting to terminal */
-- 
1.7.1